File: Samples/C#.Net/CSharpSDKTest/ISmpPluginTabDlg.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using SalesMatePlusLib;
using System.IO;
using System.Reflection;

namespace CSharpSDKTest
{
    public partial class ISmpPluginTabDlg : Form
    {
        int iView;
        int iFormView;
        ISmpPluginTab m_SmpPluginTab = new SmpPluginTab();


        public ISmpPluginTabDlg()
        {
            InitializeComponent();
            iView = 0;
            iFormView = 0;
        }

        private void buttonCreateView_Click(object sender, EventArgs e)
        {
            object vTabView = new object();
            vTabView = null;
            int m_lIndex=-1;
            iView = iView + 1;
            int iCount;
            iCount = iView;
            string strLabel = "View";
            strLabel = strLabel + iCount;
            string strToolTip ="View ToolTip";
            strToolTip = strToolTip + iCount;
            m_SmpPluginTab.CreateView(strLabel, strToolTip, ref vTabView, ref m_lIndex);
        }

        private void buttonCreateFormView_Click(object sender, EventArgs e)
        {
            object vTabView = new object();
            vTabView = null;
            int m_lIndex=-1;
            iFormView = iFormView+1;
            int iCount;
            iCount = iFormView;
            String strLabel = "FormView";
            strLabel = strLabel + iCount;
            String strToolTip = "FormView ToolTip";
            strToolTip = strToolTip + iCount;
            m_SmpPluginTab.CreateFormView(strLabel, strToolTip, ref vTabView, ref m_lIndex);
	
        }

        private void buttonCreateHTMLView_Click(object sender, EventArgs e)
        {
            
            int lIndex = 0;
            object pBrowserObj = new object();
            m_SmpPluginTab.CreateHtmlView("SalesMate + Site", "Tooltip", ref pBrowserObj, ref lIndex);

            SHDocVw.IWebBrowser2 pBrowser2 = (SHDocVw.IWebBrowser2)pBrowserObj;
            object Param1 = new object();
            object Param2= new object();
            object Param3= new object();
            object Param4 = new object();
            pBrowser2.Navigate("http://www.salesmateplus.com", ref Param1, ref Param2, ref Param3,ref Param4);
        }

        private void buttonSet_Click(object sender, EventArgs e)
        {
            try
            {

                if (textBoxCurrentView.Text == "")
                {
                    m_SmpPluginTab.SetCurrentView(0);
                }
                else
                {
                    int lIndex = Convert.ToInt32(textBoxCurrentView.Text);
                    m_SmpPluginTab.SetCurrentView(lIndex);

                }

            }
            catch {
                m_SmpPluginTab.SetCurrentView(0);
            }
        }
    }
}

File: Samples/VB.Net/VBDotNet/ISmpPluginTabDlg.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports SalesMatePlusLib

Public Class ISmpPluginTabDlg

    Dim iView As Integer
    Dim iFormView As Integer
    Dim m_SmpPluginTab As SmpPluginTab = New SmpPluginTab()

    Public Sub New()
        ' This call is required by the Windows Form Designer.
        InitializeComponent()
        iView = 0
        iFormView = 0
    End Sub

    Private Sub btnCreateView_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCreateView.Click
        Dim vTabView As New Object()
        vTabView = Nothing
        Dim m_lIndex As Integer = -1
        iView = iView + 1
        Dim strCount As String
        Dim strView As String
        strView = (Convert.ToInt32(iView)).ToString()
        strCount = strView
        Dim strLabel As String = "View"
        strLabel = strLabel + strCount
        Dim strToolTip As String = "View ToolTip"
        strToolTip = strToolTip + strCount
        m_SmpPluginTab.CreateView(strLabel, strToolTip, vTabView, m_lIndex)
    End Sub

    Private Sub btnCreateFormView_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCreateFormView.Click
        Dim vTabView As New Object()
        vTabView = Nothing
        Dim m_lIndex As Integer = -1
        iFormView = iFormView + 1
        Dim iCount As Integer
        Dim strCount As String
        strCount = (Convert.ToInt32(iCount)).ToString()
        iCount = iFormView
        Dim strLabel As [String] = "FormView"
        strLabel = strLabel + strCount
        Dim strToolTip As [String] = "FormView ToolTip"
        strToolTip = strToolTip + strCount
        m_SmpPluginTab.CreateFormView(strLabel, strToolTip, vTabView, m_lIndex)
    End Sub

    Private Sub btnCreateHtmlView_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCreateHtmlView.Click

        Dim lIndex As Integer = 0
        Dim pBrowserObj As New Object()
        m_SmpPluginTab.CreateHtmlView("SalesMate + Site", "Tooltip", pBrowserObj, lIndex)

        Dim pBrowser2 As SHDocVw.IWebBrowser2 = DirectCast(pBrowserObj, SHDocVw.IWebBrowser2)
        Dim Param1 As New Object()
        Dim Param2 As New Object()
        Dim Param3 As New Object()
        Dim Param4 As New Object()
        pBrowser2.Navigate("http://www.salesmateplus.com", Param1, Param2, Param3, Param4)

    End Sub

    Private Sub btnSetCurrentView_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSetCurrentView.Click
        Try
            If txtSetCurrentViewTo.Text = "" Then
                m_SmpPluginTab.SetCurrentView(0)
            Else
                Dim lIndex As Integer = Convert.ToInt32(txtSetCurrentViewTo.Text)
                m_SmpPluginTab.SetCurrentView(lIndex)
            End If
        Catch ex As Exception
            m_SmpPluginTab.SetCurrentView(0)
        End Try
    End Sub
End Class